spec/requests/session_controller_spec.rb

```diff
@@ -1245,9 +1247,13 @@ def login_with_sso_and_invite(invite_key = invite.invite_key)
         expect(sso2.no_2fa_methods).to eq(nil)
       end
 
+      it "correctly logs in for secondary domain secrets" do
+        get "/session/sso_provider", params: Rack::Utils.parse_query(@sso.payload("newSecretForOverRainbow"))
+        expect(response.status).to eq(302)
+      end
```

Comment:
This isn't really testing the functionality - the 302 was to the forum login page because no user was signed in. I suggest we test both old and new secrets, and verify the return signature is correct.

```suggestion
      it "correctly logs in for secondary domain secrets" do
        sign_in @user

        get "/session/sso_provider", params: Rack::Utils.parse_query(@sso.payload("newSecretForOverRainbow"))
        expect(response.status).to eq(302)
        redirect_uri = URI.parse(response.location)
        expect(redirect_uri.host).to eq("somewhere.over.rainbow")
        redirect_query = CGI.parse(redirect_uri.query)
        expected_sig = DiscourseConnectBase.sign(redirect_query["sso"][0], "newSecretForOverRainbow")
        expect(redirect_query["sig"][0]).to eq(expected_sig)

        get "/session/sso_provider", params: Rack::Utils.parse_query(@sso.payload("oldSecretForOverRainbow"))
        expect(response.status).to eq(302)
        redirect_uri = URI.parse(response.location)
        expect(redirect_uri.host).to eq("somewhere.over.rainbow")
        redirect_query = CGI.parse(redirect_uri.query)
        expected_sig = DiscourseConnectBase.sign(redirect_query["sso"][0], "oldSecretForOverRainbow")
        expect(redirect_query["sig"][0]).to eq(expected_sig)
      end
```